src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/AttackSystem.java

```diff
@@ -75,35 +47,12 @@ private void dropFlagOnPlayerAttack(ActivateEvent event, EntityRef targetPlayer)
         EntityRef attackingPlayer = event.getInstigator(); // The player using the staff to attack
         if (canPlayerAttack(attackingPlayer)) {
             if (targetPlayer.hasComponent(PlayerCharacterComponent.class) && targetPlayer.hasComponent(HasFlagComponent.class)) {
-                // If the target player has the black flag
-                if (targetPlayer.getComponent(HasFlagComponent.class).flag.equals(LASUtils.BLACK_TEAM)) {
-                    dropFlag(targetPlayer, attackingPlayer, LASUtils.BLACK_FLAG_URI);
-                    return;
-                }
-                if (targetPlayer.getComponent(HasFlagComponent.class).flag.equals(LASUtils.RED_TEAM)) {
-                    dropFlag(targetPlayer, attackingPlayer, LASUtils.RED_FLAG_URI);
-                    return;
-                }
+                // If the target player has the flag
+                targetPlayer.send(new DropFlagEvent(attackingPlayer));
```

Comment:
why do we send the attacking player here?

Reply:
I guess we could decide to compute the direction from which the magic staff "activation" comes here in the attack system and only pass that to the `DropFlagEvent`. 

On the other hand, this might be a nice addition to the score board if you think for "scorer points" from other sports, e.g., giving players credit for defending their flag...